identity-account/src/updates/update.rs

```diff
@@ -102,7 +102,7 @@ pub(crate) async fn create_identity(
 }
 
 #[derive(Clone, Debug)]
-pub(crate) enum Update {
+pub enum Update {
   CreateMethod {
     scope: MethodScope,
     type_: MethodType,
```

Comment:
Do we really need this public? This significantly affects our Rust users in my opinion since it introduces another method of updating credentials apart from the builder.

Can the Wasm bindings not just call the `update_identity()` builders internally without exposing this?

Reply:
The advantage here is that all properties must be exhausted in the bindings, which is important not to forget any of them if we use the builder. But you have a good point here. I can change it if we decide that it's not worth it. But I personally don't see a big issue. Plus, I really dislike not having auto completion for the builder (I think it's only possible when using the Rust Analyzer, which I don't use) because it's implemented in a macro.

Reply:
Yeah, we've had discussions and I've been thinking more about this and we might want to expose the update structs even in Rust. Will need to be a joint decision though.

Reply:
Pros:
- Changing an `Update` variant more likely produces a compiler error in the Wasm bindings, e.g. if a field changes from optional to required (unlikely, imo).

Cons:
- Exposes a second method of updating the account which pollutes the API unnecessarily on the Rust side.
- Duplicates default logic (very minor, only affects `Update::CreateMethod::type_`, unlikely to change).

We could make `process_update` public behind the `wasm` feature, but I don't really like that option either, just adds to the fragmentation we will have with `Storage`. I think using the builders is better, as a change to the `Update` enum will most likely require changes to Wasm as well, and would likely also produce a compiler error, and so I find it unlikely that some change slips through unnoticed.